### ISSUE TITLE
Convert LinkedList to ArrayList or ArrayDeque while possible to improve performance.

### ISSUE THREAD

**Issue by huizhilu** [apache/helix#350]
I find that some code is still using LinkedList instead of ArrayList for the List object. Ex.:
1) https://github.com/apache/helix/blob/323fbd049ed84a17ca8a2a00019bf76d51f5346e/helix-admin-webapp/src/main/java/org/apache/helix/webapp/resources/InstancesResource.java#L94
2) https://github.com/apache/helix/blob/c98dfa056e8ff67666b90827f369d74eea8213bf/helix-core/src/main/java/org/apache/helix/common/caches/ExternalViewCache.java#L98

I believe ArrayList has better performance over LinkedList in most cases, especially when the dataset is big and we don't need to randomly delete data when using an iterator.

If LinkedList is used as a queue, we might want to look at Deque/ArrayDeque, which has better performance over LinkedList. ex.:
https://github.com/apache/helix/blob/323fbd049ed84a17ca8a2a00019bf76d51f5346e/helix-core/src/main/java/org/apache/helix/tools/commandtools/ZKLogFormatter.java#L117

In summary, we should avoid using LinkedList as far as possible. Maybe for small dataset, linkedlist and ArrayList have similar performance. If we only target at small dataset in this framework, it should be fine. But how about using ArrayDeque over LinkedList for a Queue implementation?

References: 
https://stackoverflow.com/questions/6163166/why-is-arraydeque-better-than-linkedlist
http://brianandstuff.com/2016/12/12/java-arraydeque-vs-linkedlist/
